### PR TITLE
perf: parallelize swc js minify

### PR DESCRIPTION
**Pull request by h-a-n-a**
## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Comment by h-a-n-a** [comment]
!bench

**Comment by h-a-n-a** [comment]
Performance is related to assets generated, so this won't be a huge performance leap for the benchmark. But still we can have a look.

**Comment by h-a-n-a** [comment]
!bench

File: crates/rspack_plugin_swc_js_minimizer/src/lib.rs [hunk 1]
```diff
@@ -64,12 +59,11 @@ pub enum SwcJsMinimizerRules {
...
 
 impl SwcJsMinimizerRules {
-  #[async_recursion]
-  pub async fn try_match(&self, data: &str) -> rspack_error::Result<bool> {
+  pub fn try_match(&self, data: &str) -> rspack_error::Result<bool> {
     match self {
       Self::String(s) => Ok(data.starts_with(s)),
       Self::Regexp(r) => Ok(r.test(data)),
-      Self::Array(l) => try_any(l, |i| async { i.try_match(data).await }).await,
+      Self::Array(l) => try_any_sync(l, |i| i.try_match(data)),
```
Comment (h-a-n-a) [inline crates/rspack_plugin_swc_js_minimizer/src/lib.rs:66]: Let's change to sync until necessary.

**Review by hyf0 — DISMISSED** [review]

**Comment by h-a-n-a** [comment]
The original implementation collects all diagnostics from the minifications and report all once. Let's keep the semantics.

**Review by ahabhgk — APPROVED** [review]

**Comment by h-a-n-a** [comment]
!canary

**Pull request merged**